llm_toolkit/prompt_builder.py

```diff
@@ -543,3 +548,116 @@ def build_fixer_prompt(self, benchmark: Benchmark, raw_code: str,
     """Builds a fixer prompt."""
     # Do nothing for jvm project now.
     return self._prompt
+
+
+class CSpecificBuilder(PromptBuilder):
+  """Builder specifically targeted C (and excluding C++)."""
+
+  def __init__(self,
+               model: models.LLM,
+               benchmark: Benchmark,
+               template_dir: str = DEFAULT_TEMPLATE_DIR):
+    super().__init__(model)
+    self._template_dir = template_dir
+    self.benchmark = benchmark
+
+    # Load templates.
+    self.priming_template_file = self._find_template(template_dir,
+                                                     'c-priming.txt')
```

Comment:
> 1. Does this template perform better than prompts/template_xml/priming.txt? (If so, I will modify and use it for C++).

I don't think of it as necessarily better or worse. I came up with the text based on what I find intuitive when building a harness. The two texts to me are both valuable and I can likely see one producing better harnesses in one scenario and the other building better harnesses in another scenario. I think there should be a larger spread to account for prompts not necessarily being a 1-dimensional comparison (which is better/worse) but rather a multi-dimensional (x performs better in scenarios m,z,v and y performs better in a,b,c).

> 2. I appreciate that you use a new prompt to avoid being intrusive. 

Not really -- the PR is made to be non-intrusive, but the prompt itself was build with more in mind. I don't subscribe to one-prompt-to-rule-them-all at this point, but rather that different prompts will perform different in a case of many diverse targets. So the effort with a new prompt is also about opening the door to that.

I would prefer to maintain two different prompts -- and potentially more if/when some come up with domain-specific prompts -- primarily for this reason.

I also think the existing prompt is getting fairly complex and is a bit difficult to read. I would prefer not to make it any more complex.